chcemvediet/apps/inforequests/tests/test_models/test_action.py

```diff
@@ -0,0 +1,874 @@
+
+    def test_is_applicant_action_is_obligee_action_and_is_implicit_action_properties(self):
+        tests = (                                   # Applicant, Obligee, Implicit
+                (Action.TYPES.REQUEST,                True,      False,   False),
+                (Action.TYPES.CLARIFICATION_RESPONSE, True,      False,   False),
+                (Action.TYPES.APPEAL,                 True,      False,   False),
+                (Action.TYPES.CONFIRMATION,           False,     True,    False),
+                (Action.TYPES.EXTENSION,              False,     True,    False),
+                (Action.TYPES.ADVANCEMENT,            False,     True,    False),
+                (Action.TYPES.CLARIFICATION_REQUEST,  False,     True,    False),
+                (Action.TYPES.DISCLOSURE,             False,     True,    False),
+                (Action.TYPES.REFUSAL,                False,     True,    False),
+                (Action.TYPES.AFFIRMATION,            False,     True,    False),
+                (Action.TYPES.REVERSION,              False,     True,    False),
+                (Action.TYPES.REMANDMENT,             False,     True,    False),
+                (Action.TYPES.ADVANCED_REQUEST,       False,     False,   True),
+                (Action.TYPES.EXPIRATION,             False,     False,   True),
+                (Action.TYPES.APPEAL_EXPIRATION,      False,     False,   True),
+                )
+        # Make sure we are testing all defined action types
+        tested_action_types = [a for a, _, _, _ in tests]
+        defined_action_types = Action.TYPES._inverse.keys()
+        self.assertItemsEqual(tested_action_types, defined_action_types)
+
+        for action_type, is_applicant, is_obligee, is_implicit in tests:
+            action = self._create_action(type=action_type)
+            self.assertEqual(action.is_applicant_action, is_applicant)
+            self.assertEqual(action.is_obligee_action, is_obligee)
+            self.assertEqual(action.is_implicit_action, is_implicit)
+
+    def test_is_by_email_property(self):
+        email = self._create_message()
+        action1 = self._create_action(email=email)
+        action2 = self._create_action()
+        self.assertTrue(action1.is_by_email)
+        self.assertFalse(action2.is_by_email)
+
+    @contextlib.contextmanager
+    def _test_deadline_aux(self, **kwargs):
+        action = self._create_action(legal_date=naive_date(u'2010-10-05'), delivered_date=naive_date(u'2010-10-05'), **kwargs)
+        with mock.patch(u'chcemvediet.apps.inforequests.models.action.workdays.between', side_effect=lambda a,b,*args: (b-a).days):
+            yield action
+
+    def test_deadline_workdays_passed_property_and_deadline_workdays_passed_at_method(self):
+        with self._test_deadline_aux() as action:
+            timewarp.jump(local_datetime_from_local(u'2010-10-10 10:33:00'))  # 5 days behind base_date
```

Comment:
@viliambalaz Radsej do tych komentarov napis, co presne kde je ten `base_date`. Ci `legal_date`, ci `delivered_date` alebo co.

Reply:
@martinmacko47 tieto komenty vymazani `test_deadline_missed_aux` stratili opodstatnenie. Vtedy som chcel poukazat aky hodnota bola `legal_date`/ `base_date`.  Teraz sa to da lahko precitat, kedze pouzivame `_create_action`